common/short-hex-str/src/lib.rs

```diff
@@ -20,7 +20,7 @@ pub struct InputTooShortError;
 
 impl ShortHexStr {
     pub const SOURCE_LENGTH: usize = 4;
-    pub const LENGTH: usize = 2 * ShortHexStr::SOURCE_LENGTH;
+    pub const LENGTH: usize = ShortHexStr::SOURCE_LENGTH.saturating_mul(2);
```

Comment:
shouldn't need this; the compiler will actually error if multiplying your constants will overflow. this just makes it harder to read for no gain.

Reply:
does it ? As far as I knew, the compiler will only error in debug mode on overflows. but in normal mode, it just currently wraps around. @mimoo  can you confirm?

Reply:
@xvschneider philip is saying that the compiler should detect the overflow (if there is one) at compile time since these are constants. Unfortunately this is a bug in clippy, there's a task for that but I don't have the link anymore. We shouldn't be fixing them, clippy should fix the lint instead : o

Reply:
here's a playground link showing the compile error: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=242804d1e73e76ed3e09c66abdc57561

```
error: this arithmetic operation will overflow
  --> src/main.rs:10:20
   |
10 |     println!("{}", a * b);
   |                    ^^^^^ attempt to compute `123_u8 * 3_u8`, which would overflow
   |
   = note: `#[deny(arithmetic_overflow)]` on by default
```

Reply:
makes sense. in this case I added "allow" for integer arithmetic until the lint is fixed. thanks for the clarification @phlip9  and @mimoo !